tests/by-util/test_nl.rs

```diff
@@ -800,3 +800,102 @@ fn test_file_with_non_utf8_content() {
         String::from_utf8_lossy(invalid_utf8)
     ));
 }
+
+// Regression tests for issue #9132: repeated flags should use last value
+#[test]
+fn test_repeated_body_numbering_flag() {
+    // -ba -bt should use -bt (number all lines)
+    new_ucmd!()
+        .args(&["-ba", "-bt"])
+        .pipe_in("a\nb\nc")
+        .succeeds()
+        .stdout_is("     1\ta\n     2\tb\n     3\tc\n");
+}
+
+#[test]
+fn test_repeated_header_numbering_flag() {
+    // -ha -ht should use -ht (number all lines in header)
+    new_ucmd!()
+        .args(&["-ha", "-ht"])
+        .pipe_in("a\nb\nc")
```

Comment:
In addition to what I previously wrote about the missing empty line in the input, the input here is not a header and thus `-h` has no effect. A header starts with `\:\:\:\n` by default.